src/vsphere_cpi/spec/unit/cloud/vsphere/nsxt_provider_spec.rb

```diff
@@ -93,16 +96,17 @@
   let(:default_vif_type) { 'PARENT' }
 
   subject(:nsxt_provider) do
-    described_class.new(client, default_vif_type).tap do |provider|
+    described_class.new(client, default_vif_type, vcenter_client, datacenter).tap do |provider|
       provider.instance_variable_set('@client', client)
     end
   end
 
   describe '#set_vif_type' do
     let(:success_response) { HTTP::Message.new_response('') }
-
     before do
       allow(nsxt_provider).to receive(:logical_ports).with(vm).and_return([logical_port_1])
+      allow(nsxt_provider).to receive(:nsxt_nics).with(vm).and_return(nic)
+      allow(nic).to receive(:empty).and_return(false)
```

Comment:
Hmm what's this?

Reply:
Every method checks `nsxt_nics(vm).empty?` even before attempting to find logical ports and moving ahead. I tried to re-use this pre-condition check using the `before` block for all the tests within the block.

Reply:
No need to do this since `nic` is already an `Array` (actually it should probably be called `nics`) and will respond to `empty?` correctly. Also this line doesn't actually do anything right now. I think this was intended:

```ruby
allow(nic).to receive(:empty?).and_return(false)
```

Note the question mark. But either way it's unnecessary.